.github/workflows/ci.yml

```diff
@@ -0,0 +1,127 @@
+name: Continuous Integration
+
+on:
+  push:
+    branches:
+    - master
+
+  pull_request:
+    branches:
+    - master
+
+jobs:
+  coverage:
+    runs-on: ubuntu-latest
+
+    # To enable coverage, go to https://codecov.io/, acquire a token, put it
+    # into your secrets (Settings > Secrets > New repository secret) and delete
+    # the "if" conditional below.
+    # If you do not wish to use codecov, then simply delete this job from the
+    # workflow.
+    if: "false"
+
+    steps:
+    - uses: actions/checkout@v1
+
+    - name: Install LCov
+      run: sudo apt-get update -q
+        && sudo apt-get install lcov -q -y
+
+    - name: Install vcpkg
+      uses: lukka/run-vcpkg@v7
+      with: 
+        setupOnly: true
+        vcpkgGitCommitId: 7c55ecac266fc3e554bc315053dc45c11ec811af
+
+    - name: Configure
+      run: cmake --preset=ci-coverage
+        -D "VCPKG_MANIFEST_FEATURES=test;magic-enum"
+
+    - name: Build
+      run: cmake --build build -j 2
+
+    - name: Test
+      working-directory: build
+      env:
+        CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
+        PS4: "\e[36m+++\e[0m "
+      run: |
+        set -x
+        ctest --output-on-failure -j 2
+        lcov -c -d .. -o coverage.info --include '${{ github.workspace }}/*'
+        lcov --list coverage.info
+        bash <(curl -s https://codecov.io/bash) -f coverage.info
```

Comment:
Great! Can you expose a cmake target for doing the coverage report locally also?

Reply:
I found that CTest's coverage support is not quite as good as the lcov generated document, so I left this as just a shell script. Since this is not cross-platform, I don't know what'd be the best way to have it in the CMake scripts. I have to think about this for cmake-init as well.

Reply:
I think you can implement any command in cmake using something like [add_custom_command](https://cmake.org/cmake/help/latest/command/add_custom_command.html) ? 
It doesnt have to be cross platform, we can mention it in the documentation that it is only works for ubuntu-latest.
I would like to be able to see coverage locally, since I had that before with gcov in the Makefile.
Would you like to take on the exercise? Then you can back-port the solution to cmake-init

Reply:
I made an issue for it: https://github.com/cdeln/cpp_enum_set/issues/4